File: Repository/ParticipantDBRepository.cs
using log4net;
using Motociclete.Domain;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Microsoft.Data.Sqlite;

namespace Motociclete.Repository
{
    public class ParticipantDBRepository : IRepositoryParticipant
    {
        private static readonly ILog logger = LogManager.GetLogger(typeof(ParticipantDBRepository));
        private readonly DBUtils _dbUtils;
        private readonly EchipaDBRepository _echipaDBRepository;

        public ParticipantDBRepository(IDictionary<string, string> properties, EchipaDBRepository echipaDBRepository)
        {
            _dbUtils = new DBUtils(properties);
            _echipaDBRepository = echipaDBRepository;
            logger.Info("ParticipantDBRepository initialized");
        }

        public long Insert(Participant entity)
        {
            logger.Info("Inserting new Participant");

            using (IDbConnection conn = _dbUtils.getConnection())
            {
                try
                {
                    if (conn is SqliteConnection sqliteConn)
                    {
                        string query = "INSERT INTO Participant (nume, capMotor, echipa, cnp) VALUES (@nume, @capMotor, @echipa, @cnp); SELECT last_insert_rowid();";

                        using (var cmd = new SqliteCommand(query, sqliteConn))
                        {
                            cmd.Parameters.AddWithValue("@nume", entity.Nume);
                            cmd.Parameters.AddWithValue("@capMotor", entity.CapMotor);
                            cmd.Parameters.AddWithValue("@echipa", entity.Echipa.id);
                            cmd.Parameters.AddWithValue("@cnp", entity.Cnp);

                            long id = (long)cmd.ExecuteScalar();
                            logger.Info($"Inserted new Participant with id {id}");
                            return id;
                        }
                    }
                }
                catch (Exception ex)
                {
                    logger.Error("Error inserting Participant: " + ex.Message);
                }
            }
            return -1;
        }

        public void UpdateById(long id, Participant entity)
        {
            throw new NotImplementedException();
        }

        public void DeleteById(long id)
        {
            logger.Info($"Deleting Participant with id {id}");

            using (IDbConnection conn = _dbUtils.getConnection())
            {
                try
                {
                    if (conn is SqliteConnection sqliteConn)
                    {
                        string query = "DELETE FROM Participant WHERE id = @id";

                        using (var cmd = new SqliteCommand(query, sqliteConn))
                        {
                            cmd.Parameters.AddWithValue("@id", id);
                            cmd.ExecuteNonQuery();
                            logger.Info("Participant deleted successfully");
                        }
                    }
                }
                catch (Exception ex)
                {
                    logger.Error("Error deleting Participant: " + ex.Message);
                }
            }
        }

        public Participant GetById(long id)
        {
            logger.Info($"Fetching Participant with id {id}");

            using (IDbConnection conn = _dbUtils.getConnection())
            {
                try
                {
                    if (conn is SqliteConnection sqliteConn)
                    {
                        string query = "SELECT * FROM Participant WHERE id = @id";

                        using (var cmd = new SqliteCommand(query, sqliteConn))
                        {
                            cmd.Parameters.AddWithValue("@id", id);
                            using (var reader = cmd.ExecuteReader())
                            {
                                if (reader.Read())
                                {
                                    var echipa = _echipaDBRepository.GetById(reader.GetInt64(3));
                                    return new Participant(reader.GetInt64(0), reader.GetString(1), reader.GetInt32(2), echipa, reader.GetString(4));
                                }
                            }
                        }
                    }
                }
                catch (Exception ex)
                {
                    logger.Error("Error fetching Participant: " + ex.Message);
                }
            }
            return null;
        }

        public List<Participant> GetAll()
        {
            logger.Info("Fetching all Participants");

            List<Participant> participants = new List<Participant>();

            using (IDbConnection conn = _dbUtils.getConnection())
            {
                try
                {
                    if (conn is SqliteConnection sqliteConn)
                    {
                        string query = "SELECT * FROM Participant";

                        using (var cmd = new SqliteCommand(query, sqliteConn))
                        {
                            using (var reader = cmd.ExecuteReader())
                            {
                                while (reader.Read())
                                {
                                    var echipa = _echipaDBRepository.GetById(reader.GetInt64(3));
                                    participants.Add(new Participant(reader.GetInt64(0), reader.GetString(1), reader.GetInt32(2), echipa, reader.GetString(4)));
                                }
                            }
                        }
                    }
                }
                catch (Exception ex)
                {
                    logger.Error("Error fetching all Participants: " + ex.Message);
                }
            }
            return participants;
        }
    }

}

File: RaceAppJava/Persistence/src/main/java/org/example/ParticipantDBRepository.java
package org.example;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.model.Participant;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.Properties;

import java.util.List;

public class ParticipantDBRepository implements ParticipantRepositoryInterface {
    private static final Logger logger = LogManager.getLogger(CursaDBRepository.class);
    private JdbcUtils dbUtils;

    public ParticipantDBRepository(Properties properties) {
        logger.info("Initializing AngajatDBRepository with properties: {}", properties);
        dbUtils = new JdbcUtils(properties);
    }

    @Override
    public Long insert(Participant entity) {
        logger.traceEntry("Inserting new Participant");
        Connection conn = dbUtils.getConnection();

        try (PreparedStatement prepStmt = conn.prepareStatement("INSERT INTO Participant(nume, capMotor, echipa, cnp) VALUES (?,?,?,?)")) {
            prepStmt.setString(1, entity.getNume());
            prepStmt.setInt(2, entity.getCapMotor());
            prepStmt.setString(3, entity.getEchipa());
            prepStmt.setString(4, entity.getCnp());
            prepStmt.executeUpdate();
            ResultSet rs = prepStmt.getGeneratedKeys();
            if (rs.next()) {
                logger.traceExit("Inserted new Participant with id{}", rs.getLong(1));
                return rs.getLong(1);
            }
        } catch (SQLException ex) {
            logger.error("error at insertion: " + ex);
            System.out.println("DB error: " + ex);
        }
        return null;
    }

    @Override
    public void updateById(Long aLong, Participant entity) {

    }

    @Override
    public void deleteById(Long aLong) {

    }

    @Override
    public Participant getById(Long aLong) {
        return new Participant(1L,"err",1234,"err","err");
    }

    @Override
    public List<Participant> getAllByTeam(String team) {
        logger.info("Getting all Participants by team {}", team);
        Connection conn = dbUtils.getConnection();
        List<Participant> participants = new ArrayList<>();

        try (PreparedStatement prepStmt = conn.prepareStatement("SELECT * FROM participant WHERE echipa = ?")) {
            prepStmt.setString(1, team);
            try (ResultSet rs = prepStmt.executeQuery()) {
                while (rs.next()) {
                    participants.add(new Participant(
                            rs.getLong("id"),
                            rs.getString("nume"),
                            rs.getInt("capMotor"),
                            rs.getString("echipa"),
                            rs.getString("cnp")
                    ));
                }
            }
        } catch (SQLException e) {
            logger.error("Error fetching all Participants by team: " + e.getMessage());
        }
        return participants;
    }

    @Override
    public List<Participant> getAll() {
        logger.info("Fetching all Participants");
        Connection conn = dbUtils.getConnection();
        List<Participant> participants = new ArrayList<>();

        try (PreparedStatement prepStmt = conn.prepareStatement("SELECT * FROM participant");
             ResultSet rs = prepStmt.executeQuery()) {
            while (rs.next()) {
                participants.add(new Participant(
                        rs.getLong("id"),
                        rs.getString("nume"),
                        rs.getInt("capMotor"),
                        rs.getString("echipa"),
                        rs.getString("cnp")
                ));
            }
        } catch (SQLException e) {
            logger.error("Error fetching all Participants: " + e.getMessage());
        }
        return participants;
    }

    @Override
    public List<String> getEchipe() {
        logger.info("Fetching all team names");
        Connection conn = dbUtils.getConnection();
        List<String> echipe = new ArrayList<>();

        try (PreparedStatement prepStmt = conn.prepareStatement("SELECT DISTINCT echipa FROM participant");
             ResultSet rs = prepStmt.executeQuery()) {
            while (rs.next()) {
                echipe.add(rs.getString("echipa"));
            }
        } catch (SQLException e) {
            logger.error("Error fetching team names: " + e.getMessage());
            throw new RuntimeException("Error fetching team names");
        }
        return echipe;
    }
}
